### PR TITLE
[souschef] Introduce Dims

### PR DESCRIPTION
**Pull request by seanshpark**
This will introduce Dims related helpers, common of circlechef and tflchef

ONE-DCO-1.0-Signed-off-by: SaeHie Park <saehie.park@gmail.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
For #3881 from draft #3905

File: compiler/souschef/src/Dims.cpp [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "souschef/Dims.h"
+
+// NOTE Do NOT delete this file; this file checks the completeness of 'Dims.h'
```
Comment (mhs4670go) [inline compiler/souschef/src/Dims.cpp:19]: Could you explain about this more detail? I mean what the completeness you said is.

Comment (seanshpark) [reply compiler/souschef/src/Dims.cpp:19]: Like other header only files, (you can search `compiler` folder with this string), this cpp will include the header `Dims.h` and compile.
If not, `Dims.h` won't be compiled and CI won't tell if there is compilation error.

**Review by mhs4670go — APPROVED** [review]
LGTM

**Pull request merged**